langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/FromJsonWithType.java

```diff
@@ -124,6 +125,9 @@ private static Object convert(Object value, Type targetType, List<TypeValuePair>
             case TypeTags.TUPLE_TAG:
                 newValue = convertArray((BArray) value, matchingType, unresolvedValues, t);
                 break;
+            case TypeTags.REG_EXP_TYPE_TAG:
+                newValue = RegExpFactory.parse(((BString) value).getValue());
```

Comment:
Shouldn't we have a try-catch here and return ballerina error since we are only catching BallerinaException?